net/levin/src/message.rs

```diff
@@ -0,0 +1,198 @@
+//! Levin Messages
+//!
+//! This module contains the [`LevinMessage`], which allows sending bucket body's, full buckets or dummy messages.
+//! The codec will not return [`LevinMessage`] instead it will only return bucket body's. [`LevinMessage`] allows
+//! for more control over what is actually sent over the wire at certain times.
+use bytes::{Bytes, BytesMut};
+
+use crate::{
+    header::{Flags, HEADER_SIZE},
+    Bucket, BucketBuilder, BucketError, BucketHead, LevinBody, LevinCommand, Protocol,
+};
+
+/// A levin message that can be sent to a peer.
+pub enum LevinMessage<T: LevinBody> {
+    /// A message body.
+    /// A levin header will be added to this message before it is sent to the peer.
+    Body(T),
+    /// A full levin bucket.
+    /// This bucket will be sent to the peer directly with no extra information.
+    /// This should only be used to send fragmented messages: [`make_fragmented_messages`]
+    Bucket(Bucket<T::Command>),
+    /// A Dummy message
+    /// A dummy message which the peer will ignore. The dummy message will be the exact size
+    /// (in bytes) of the given `usize` on the wire.
+    Dummy(usize),
+}
+
+impl<T: LevinBody> From<T> for LevinMessage<T> {
+    fn from(value: T) -> Self {
+        LevinMessage::Body(value)
+    }
+}
+
+impl<T: LevinBody> From<Bucket<T::Command>> for LevinMessage<T> {
+    fn from(value: Bucket<T::Command>) -> Self {
+        LevinMessage::Bucket(value)
+    }
+}
+
+/// This represents a dummy message to send to a peer.
+///
+/// The message, including the header, will be the exact size of the given `usize`.
+// This exists because it seems weird to do this:
+// peer.send(1_000);
+// This is a lot clearer:
+// peer.send(Dummy(1_000));
```

Comment:
```suggestion
/// This represents a dummy message to send to a peer.
///
/// The message, including the header, will be the exact size of the given `usize`.
/// This exists because it seems weird to do this:
/// ```rust,ignore
/// peer.send(1_000);
/// ```
/// This is a lot clearer:
/// ```rust,ignore
/// peer.send(Dummy(1_000));
/// ```
```